### ISSUE TITLE
Remove JSON.Net dependency

### ISSUE THREAD

**Issue by dylanvdmerwe** [Redth/PushSharp#663]
### What version of PushSharp are you using?
4.x

### Describe your issue:
Is there really a need to have a dependency on the JSON.Net library anymore now that developers are responsible for creating their own message payloads?

Currently for GCM and Apns:
![1](https://cloud.githubusercontent.com/assets/2852839/14012598/62fe0c8c-f1af-11e5-9bbb-71c2cfefd3af.png)

![2](https://cloud.githubusercontent.com/assets/2852839/14012512/d5a912fa-f1ae-11e5-8695-4390daf481d8.png)

The Data and Payload properties respectively expect a JObject. This creates a hard dependency on the JSON.Net library. 

I firmly believe it should be up to the developers to create a proper payload and that it is their responsibility to serialize the JSON correctly. As such, the Notification classes should just take a string object instead of a JObect. 

Benefits of this:

- Removing hard dependencies on external libs. Reduces version conflicts, binary size, etc. 
- Allows developers to make use of their own serializers like ServiceStack or Jil (or JSON.Net if they so chose)

**Comment by Redth** [comment]
Yeah i've been giving this some thought.  To be honest the only real reason it was included in 3.x+ was in case the fluent helper methods made it back in, but they have not, and will not, so I think your proposal makes sense.
